### PR TITLE
[Category Sorting] sort categories the same way as Pimcore does

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
Either by key or by index. Also check if index-sorting is already implemented

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Fixed tickets | #460 

bildschirmfoto 2018-05-28 um 22 12 49
bildschirmfoto 2018-05-28 um 22 14 04

### CODE REVIEW

File: src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php [hunk 1]
```diff
@@ -46,13 +46,37 @@ public function findChildCategoriesForStore(CategoryInterface $category, StoreIn
         $list = $this->getList();
         $list->setCondition('parentCategory__id = ? AND stores LIKE "%,'.$store->getId().',%"', [$category->getId()]);
 
+        //TODO: fix as soon as CoreShop requires pimcore/core-version:~5.2.2 as minimum
+        if (method_exists($category, 'getChildrenSortBy')) {
+            $list->setOrderKey(
+                sprintf('o_%s ASC', $category->getChildrenSortBy()),
+                false
+            );
+        }
```
Comment (solverat) [inline src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php]: formatting?

File: src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php [hunk 2]
```diff
@@ -84,6 +108,20 @@ public function findRecuriveChildCategoriesForStore(CategoryInterface $category,
...
         $list->setCondition('oo_id IN ('.implode(',', $childIds).')');
 
+        //TODO: fix as soon as CoreShop requires pimcore/core-version:~5.2.2 as minimum
+        if (method_exists($category, 'getChildrenSortBy')) {
+            $list->setOrderKey(
+                sprintf('o_%s ASC', $category->getChildrenSortBy()),
+                false
+            );
+        }
+        else {
```
Comment (solverat) [inline src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php]: formatting?

File: src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php [hunk 3]
```diff
@@ -53,18 +62,28 @@ public function findChildCategoriesForStore(CategoryInterface $category, StoreIn
      * {@inheritdoc}
      */
     public function findRecuriveChildCategoriesForStore(CategoryInterface $category, StoreInterface $store)
+    {
```
Comment (solverat) [inline src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php]: `findRecuriveChildCategoriesForStore()` isn't available in current master.

Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php]: cause it got removed by another PR

File: src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php (hunk 1 above)
Comment (dpfaffenbauer) [reply src/CoreShop/Bundle/CoreBundle/Pimcore/Repository/CategoryRepository.php]: what do you mean here?

**Comment by dpfaffenbauer** [comment]
merge conflict is resolved as well

**Pull request merged**